src/bots/liquidator.ts

```diff
@@ -685,248 +907,161 @@ export class LiquidatorBot implements Bot {
 		}
 	}
 
-	/**
-	 * attempts to close out any open positions on this account. It starts by cancelling any open orders
-	 */
-	private async derisk() {
-		if (Atomics.compareExchange(this.deriskMutex, 0, 0, 1) === 1) {
-			return;
+	private getOrderParamsForSpotDerisk(
+		subaccountId: number,
+		position: SpotPosition
+	):
+		| { tokenAmount: BN; limitPrice: BN; direction: PositionDirection }
+		| undefined {
+		const spotMarket = this.driftClient.getSpotMarketAccount(
+			position.marketIndex
+		);
+		if (!spotMarket) {
+			logger.error(
+				`failed to get spot market account for market ${position.marketIndex}`
+			);
+			return undefined;
 		}
 
-		if (!this.SELL_OPEN_POSITIONS) {
-			return;
+		const nowSec = Math.floor(Date.now() / 1000);
+		let tokenAmount = getSignedTokenAmount(
+			getTokenAmount(position.scaledBalance, spotMarket, position.balanceType),
+			position.balanceType
+		);
+		if (this.isTwapping()) {
+			const twapProgress = this.twapExecutionProgresses.get(
+				this.getTwapProgressKey(
+					MarketType.SPOT,
+					subaccountId,
+					position.marketIndex
+				)
+			);
+			twapProgress.updateProgress(tokenAmount, nowSec);
+			tokenAmount = twapProgress.getExecutionSlice(nowSec);
+			logger.info(
+				`twap progress for SPOT subaccount ${subaccountId} for market ${
+					position.marketIndex
+				}, tradeSize: ${tokenAmount.toString()}, amountRemaining: ${twapProgress
+					.getAmountRemaining()
+					.toString()}`
+			);
 		}
 
-		try {
-			const userAccount = this.driftClient.getUserAccount();
-			if (!userAccount) {
-				logger.error('failed to get user account');
-				return;
-			}
+		// need to standardize token amount to check if its closable via market orders
+		tokenAmount = standardizeBaseAssetAmount(
```

Comment:
i feel like it'd be better to not stardardize for jupiter? that way the position can actually be zero'd out?